src/dispatching/dialogue/storage/trace_storage.rs

```diff
@@ -55,10 +55,10 @@ where
         D: Send + 'static,
     {
         Box::pin(async move {
-            trace!("Updating dialogue with {}: {:#?}", chat_id, dialogue);
+            let to = format!("{:#?}", dialogue);
```

Comment:
This `format!()` allocates even if trace log level is disabled (so in most cases).

Reply:
Though it's possible to check for `log` level using https://docs.rs/log/0.4.11/log/trait.Log.html#tymethod.enabled, I don't think that it's going to be necessary, as TraceStorage only purpose is to log messages, thus using it without enabled trace log level is useless

Reply:
More like this macro:
https://docs.rs/log/0.4.11/log/macro.log_enabled.html

As I understand this, `trace` is almost always disabled and you enable it only when debugging some problem.